onnxruntime/core/framework/session_options.h

```diff
@@ -119,7 +120,13 @@ struct SessionOptions {
   std::unordered_map<std::string, const OrtValue*> initializers_to_share_map;
 
   // See onnxruntime_c_api.h for detailed documentation.
-  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val) noexcept;
+  Status AddInitializer(_In_z_ const char* name, _In_ const OrtValue* val);
+
+#if !defined(ORT_MINIMAL_BUILD)  && !defined(DISABLE_EXTERNAL_INITIALIZERS)
+  // Customer supplied pre-processed data for external initializers
+  InlinedHashMap<std::string, OrtValue> external_initializers;
```

Comment:
>external_initializers [](http://example.com/codeflow?start=40&length=21)

are users meant to call AddExternalInitializers() instead of modifying this directly? same goes for AddInitializer() and initializers_to_share_map

Reply:
AddExternalInitialiazers() includes certain checks we want to run. Same for the previous interface.

Reply:
should external_initializers and initializers_to_share_map be kept public then?